jobs/v4/howto/auto_complete_sample.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package howto
+
+// [START job_search_autocomplete_job_title]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	talent "cloud.google.com/go/talent/apiv4beta1"
+	talentpb "google.golang.org/genproto/googleapis/cloud/talent/v4beta1"
+)
+
+// jobTitleAutoComplete suggests the job titles of the given
+// company identifier on query.
+func jobTitleAutocomplete(w io.Writer, projectID, companyID, query string) (*talentpb.CompleteQueryResponse, error) {
+	ctx := context.Background()
+
+	// Initialize a completionService client.
+	c, err := talent.NewCompletionClient(ctx)
+	if err != nil {
+		fmt.Printf("talent.NewCompletionClient: %v\n", err)
+		return nil, err
+	}
+
+	// Construct a completeQuery request.
+	req := &talentpb.CompleteQueryRequest{
+		Name:        fmt.Sprintf("projects/%s", projectID),
+		Query:       query,
+		PageSize:    1,
+		CompanyName: fmt.Sprintf("projects/%s/companies/%s", projectID, companyID),
```

Comment:
Prefer to add:
scope: PUBLIC
And remove CompanyName which only works in TENANT scope, and TENANT scope is hard to test.

Reply:
@anguillanneuf Would you be OK with a follow-up PR incorporating @chenlei1216's review feedback?

Sorry I didn't reach out when Lei was added as a reviewer!

@chenlei1216 Could you TAL at each of the samples and let @anguillanneuf know when your done?  
So @anguillanneuf can make one round of replies/code changes.

If a list of samples would be useful, here are all of them (I just grep'd for START):
 - [ ] job_search_autocomplete_job_title
 - [ ] job_search_batch_delete_job
 - [ ] job_search_commute_search
 - [ ] job_search_create_client_event
 - [ ] job_search_create_company
 - [ ] job_search_create_job
 - [ ] job_search_create_job_custom_attributes
 - [ ] job_search_custom_ranking_search
 - [ ] job_search_delete_company
 - [ ] job_search_delete_job
 - [ ] job_search_get_company
 - [ ] job_search_get_job
 - [ ] job_search_histogram_search
 - [ ] job_search_list_company
 - [ ] job_search_list_job
 - [ ] job_search_quick_start_list_companies

Reply:
@chenlei1216 For this sample, in particular, should we also specify an explicit `CompletionType`?

Example sample description on website:
> _Demonstrates how to complete job titles against the public dataset of job titles_

Request:
 - query: `"partially typed job title"`
 - type: `JOB _TITLE`
 - scope: `PUBLIC`
 - page size: `1`
 - language codes: `["en"]`

The first time I used this API, I found that the various languages of the returned titles led me to specify a language (can specify many).

---

@chenlei1216 If I want to search the PUBLIC dataset, why is `name: projects/MY-PROJECT` required? The field is described as:
> _Resource name of project the completion is performed within._

But that's not exactly accurate. The completion is not being performed "within" the project.

Reply:
Hi I've reviewed everything and they look pretty good :)
For this sample in particular, I prefer to explicit set type/scope/languageCodes as you suggested, and maybe increase the pageSize to 5. 
ProjectId is always required for PUBLIC/TENANT scope so that we can easily track the customers and know who to charge. Agree that the documentation is not very accurate and we'll fix that.

Reply:
Addressed in #795